### DISCUSSION TITLE
HTTParty#lock seems to be overriding ActiveRecord’s #lock method

### DISCUSSION THREAD

**Discussion by simplecircle** [jnunemaker/httparty#740] [Q&A] [answered]
When trying to implement pessimistic locking on a Rails model I wasn't able to do so because HTTParty#lock gets called rather than the ActiveRecord method with the same name.

I'm using AR 6.1.4.1 and HTTParty 0.20.0.

`User.lock.find(1)`
`ArgumentError: wrong number of arguments (given 0, expected 1..2)
from /Users/elliottg/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/httparty-0.20.0/lib/httparty.rb:587:in `lock'`

Looks like the HTTParty code that's clobbering the AR method is within the HTTParty module line 587 as shown below.

` def lock(path, options = {}, &block)
      perform_request Net::HTTP::Lock, path, options, &block
    end`

I love HTTParty and want to keep using it.

Any ideas for a solution, other than forking? Thanks!

**Comment by simplecircle** [comment, marked as answer]
Solved. For some reason, I was explicitly including HTTParty on User, which was causing the problem.

**Reply by kabirpathak** [reply]
Hi @simplecircle, I'm facing a similar issue in my application. I have a model that uses the 'httparty' gem. Inside this when I try to create an enum, it gives me the following error : 
`You tried to define an enum named "sso_type" on the model "ReceptionSession", but this will generate a instance method "oauth?", which is already defined by another enum.`

and, 

`wrong number of arguments (given 0, expected 1..3)`

Here is my code : 

```
require 'net/http'
require 'httparty'
class Session < ActiveRecord::Base
  include HTTParty
  belongs_to :user
  has_one :acknowledgement_history

  validates :session_key, presence: true
  serialize :data

  enum sso_type: {
    oauth: 0,
    saml: 1
  }

```

On further debugging, I figured out that this is because the logger from httparty is overriding the logger from ActiveRecord::Base.

I tried removing the require statement but that did not work for me. The issue that you were facing seems similar to this one. Can you please give me a little more detail regarding what worked for you?

**Reply by jnunemaker** [reply]
Don't include httparty in an AR model. Just make a new class and include it there and use that from your AR model or whatever else you need.
